### ISSUE TITLE
Cx SideNav - Keyboard navigation doesn't work if items are rendered conditionally or async 

### ISSUE THREAD

**Issue by markywilson** [SAP/fundamental-ngx#12337]
#### Is this a bug, enhancement, or feature request?
Bug
#### Describe your proposal.
Cx SideNav - Keyboard navigation doesn't work if items are rendered conditionally or async 

#### Can you handle that on the application side
Through a workaround by calling `SideNavigationComponent.ngAfterContentInit();` after the http request is finished or the condition is fulfilled 

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
Angular 18
Fundamental 0.52

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
- import { CxSideNavigationModule } from '@fundamental-ngx/cx/side-navigation';
- Use `<fdx-side-nav>` in the html
- Add `<li fdx-nested-list-item>` using `@for` or `@if`

Expected result:
Keyboard navigation should work on all items

Current behaviour:
The keyboard navigation skips the items that are rendered conditionally or async 

#### Please provide stackblitz example(s).
https://stackblitz.com/edit/cuuove?file=src%2Fapp%2Fside-navigation-cozy-wide-text-only-1-level-example.component.ts

#### Did you check the documentation and the API?
Yes
#### Did you search for similar issues?
Yes
#### Is there anything else we should know?

**_IMPORTANT: Please refrain from providing links or screenshots of SAP's internal information, as this project is open-source, and its contents are accessible to anyone._**
